src/Build/Instance/ProjectInstance.cs

```diff
@@ -2852,7 +2906,12 @@ private void CreateEvaluatedIncludeSnapshotIfRequested(bool keepEvaluationCache,
                     }
                 }
 
-                ProjectItemInstance instance = new ProjectItemInstance(this, item.ItemType, ((IItem)item).EvaluatedIncludeEscaped, item.EvaluatedIncludeBeforeWildcardExpansionEscaped, directMetadata, inheritedItemDefinitions, item.Xml.ContainingProject.EscapedFullPath);
+                var evaluatedIncludeEscaped = ((IItem)item).EvaluatedIncludeEscaped;
+                evaluatedIncludeEscaped ??= item.EvaluatedInclude;
+                var evaluatedIncludeBeforeWildcardExpansionEscaped = item.EvaluatedIncludeBeforeWildcardExpansionEscaped;
+                evaluatedIncludeBeforeWildcardExpansionEscaped ??= item.EvaluatedInclude;
```

Comment:
Can you please add a comment to these changes? It's not clear why the fallback to `item.EvaluatedInclude` is needed.

Reply:
Added. It's for external ProjectLink based support.